### ISSUE TITLE
Missing API key and endpoint configuration in OpenAI() and support for Azure OpenAI

### ISSUE THREAD

**Issue by NaraSyuhei** [confident-ai/deepeval#1086]
Hi

Thank you for your continued efforts on this project. I’ve encountered an issue while using the following code:

https://github.com/confident-ai/deepeval/blob/dd64174925ccbacd4bb1feb332a9dcce29d87bc3/deepeval/models/gpt_model_schematic.py#L114

https://github.com/confident-ai/deepeval/blob/dd64174925ccbacd4bb1feb332a9dcce29d87bc3/deepeval/models/gpt_model_schematic.py#L132

It seems that the OpenAI() initialization is missing some necessary configurations, such as the API key and endpoint settings. Without these configurations, it doesn't seem to function as expected.

Additionally, I would like to suggest making it compatible with Azure OpenAI as well. This would broaden the utility of the library for those using Azure's services.

Could you please confirm if the OpenAI() configuration needs to include these parameters, and if there's a possibility to support Azure OpenAI in the future?

Thank you for considering this, and I appreciate your hard work!

**Comment by penguine-ip** [comment]
Hey @NaraSyuhei, have you tried setting the openai api key in the env variables?

Azure OpenAI is already supported, you can also implement your own wrappers: https://docs.confident-ai.com/docs/guides-using-custom-llms

**Comment by yfchau-news** [comment]
@penguine-ip  On a similar issue, can I use custom LLM for the synthesizer please?

**Comment by penguine-ip** [comment]
@yfchau-news definitely, also don't forget to include the custom model you want for the configs in the generation pipeline (filter config for example uses a model) https://docs.confident-ai.com/docs/synthesizer-introduction#create-your-first-synthesizer
